src/libvirtApi/domain.js

```diff
@@ -291,21 +298,56 @@ export function domainCreate({
         userLogin,
         userPassword,
         vmName,
-    });
+    };
 
     logDebug(`CREATE_VM(${vmName}): install_machine.py '${args}'`);
 
-    return cockpit
-            .spawn([pythonPath, "--", "-", args], opts)
-            .input(installVmScript)
-            .then(() => {
-                finishVmCreateInProgress(vmName, connectionName);
-                clearVmUiState(vmName, connectionName);
-            })
-            .fail(ex => {
-                clearVmUiState(vmName, connectionName); // inProgress cleanup
-                console.info(`spawn 'vm creation' returned error: "${JSON.stringify(ex)}"`);
-            });
+    const tryDownloadRhelImage = () => {
+        if (sourceType == DOWNLOAD_AN_OS && needsRHToken(os)) {
+            let arch;
+            let accessToken;
+            const options = { err: "message" };
+            if (connectionName === "system")
+                options.superuser = "try";
+
+            return cockpit.spawn(['uname', '-m'], options)
+                    .then(out => {
+                        arch = out.trim();
+
+                        return getAccessToken(offlineToken);
+                    })
+                    .then(out => {
+                        accessToken = out.trim();
+                        return getRhelImageUrl(accessToken, osVersion, arch);
+                    })
+                    .then(out => {
+                        const isSystem = connectionName === LIBVIRT_SYSTEM_CONNECTION;
+                        const outObj = JSON.parse(out);
+                        const url = outObj.url;
+                        const filename = outObj.filename;
+                        const downloadDir = isSystem ? "/var/lib/libvirt/images/" : loggedUser.home + "/.local/share/libvirt/images/";
+                        args.sourceType = LOCAL_INSTALL_MEDIA_SOURCE;
+                        args.source = downloadDir + filename;
+
+                        let buffer = "";
+                        return downloadRhelImage(accessToken, url, filename, downloadDir, isSystem)
+                                .stream(progress => {
+                                    buffer += progress;
+                                    const chunks = buffer.split("\n");
+                                    buffer = chunks.pop();
+
+                                    updateImageDownloadProgress(vmName, connectionName, chunks.pop());
```

Comment:
You need to check the length of `chunks`  here. It can happen that after pop'ing off the "rest", there is no completed chunk left over, and you get `undefined` here.